### PR TITLE
authorize /virgo/ligo FQAN to access /user/ligo and /igwn/virgo namespaces

### PR DESCRIPTION
**Pull request by AndresTanasijczuk**
I do not know why the `/virgo/ligo` FQAN is not included in the `Authorizations` list. If there is a reason, please ignore this pull request.

I would think that it should be included. It is included in the list of authorized VOMS groups for the CVMFS  repository `ligo.osgstorage.org`:

```
osg:/osg/ligo
LIGO:/ligo
virgo:/virgo/ligo
virgo:/virgo/virgo
```

So why not to add it here in topology. The `Authorizations` list here is for accessing the OSDF caches, right?

### CODE REVIEW

**Comment by brianhlin** [comment]
> The Authorizations list here is for accessing the OSDF caches, right?

Not quite: it controls access to the relevant OSDF origins, which the caches respect. I'm running this by some IGWN contacts as I'm not quite sure why it's set up like this

**Comment by AndresTanasijczuk** [comment]
> > The Authorizations list here is for accessing the OSDF caches, right?
> 
> Not quite: it controls access to the relevant OSDF origins, which the caches respect. I'm running this by some IGWN contacts as I'm not quite sure why it's set up like this

From what I can see, the SciTokens listed under Authorizations do control access to the origins and caches, but the FQANs only control access to caches, not origins. I am saying this based on what I see in the following files:

- Authfile for the LIGO VO origin 'UCL-Virgo-StashCache-Origin' serving the /user/ligo namespace: https://topology.opensciencegrid.org/stashcache/origin-authfile?fqdn=ingrid-se09.cism.ucl.ac.be. This Authfile does not contain the FQANs. It contains the DNs of the caches that are allowed to access the LIGO VO /user/ligo namespace (plus the origin itself). That is, they correspond to the caches listed under 'AllowedCaches' in https://github.com/opensciencegrid/topology/blob/master/virtual-organizations/LIGO.yaml.

- Authfile for the LIGO VO cache 'CARDIFF_UK_OSDF_CACHE' caching the /user/ligo namespace: https://topology.opensciencegrid.org/stashcache/authfile?cache_fqdn=cf-ac-uk-cache.nationalresearchplatform.org. This Authfile contains the FQANs.

- SciTokens file for the above mentioned origin: https://topology.opensciencegrid.org/stashcache/scitokens?origin_fqdn=ingrid-se09.cism.ucl.ac.be.

- SciTokens file for the above mentioned cache: https://topology.opensciencegrid.org/stashcache/scitokens?cache_fqdn=cf-ac-uk-cache.nationalresearchplatform.org.

Please correct me if I am misunderstanding something.

**Comment by brianhlin** [comment]
@AndresTanasijczuk ah, you're right! I had my head deep in the token config. I think this looks like a reasonable thing to do but need some approval from Virgo folks (I hear someone named Stefano runs the VOMS server?)

**Comment by josh-willis** [comment]
@AndresTanasijczuk Do you know the right Virgo contact for this? I asked Warren Anderson and he had suggested Stefano Bagnasco, but I know he is unavailable for an extended period.  Do you have other Virgo contacts who can weigh in? Under which circumstances are the various `/virgo`, `/virgo/virgo`, and `/virgo/ligo` certificates issued?

**Comment by AndresTanasijczuk** [comment]
The other Virgo VOMS admin with whom I always interact is @luca-rei.

As I said in the pull request comment, the reason I propose to include `/virgo/ligo` is for consistency with the authorization list in CVMFS for the ligo.osgstorage.org repository.

I do not remember what are the `/virgo`, `/virgo/virgo` and `/virgo/ligo` groups used for. Stefano Bagnasco explained it to me and Luca some time ago, but I forgot. Hopefully Luca remembers. It could be that the `/virgo/ligo` and `/virgo/virgo` groups always come together with the `/virgo` group, i.e. that they are subgroups of the `/virgo` group. If that is the case, this pull request would not be needed and the `/virgo/virgo` FQAN could be removed. But it doesn't heart to have them. 

Currently I see 8 people with the `/virgo/ligo` group in Virgo VOMS, and all of them also have the `/virgo` group.

Originally, only the `/virgo/ligo` group was included in the CVMFS ligo.osgstorage.org authorization list and only the `/virgo` group was included in topology. Luca and I requested to add `/virgo` and `/virgo/virgo` in CVMFS in https://github.com/lscsoft/osg-igwn/issues/352. This resulted in the addition of `/virgo` and `/virgo/virgo` in CVMFS and also the addition of `/virgo/virgo` in topology.

I am sorry if I am wasting your time with this. Most probably it is irrelevant.

**Comment by josh-willis** [comment]
OK thanks. @luca-rei could you add an approval to this PR? Perhaps at some point there is a discussion to be had about how many groups are really needed but that's internal to Virgo so I'll leave it to them to decide whether that needs reevaluation.

**Review by luca-rei — APPROVED** [review]
ok for me

**Review by brianhlin — APPROVED** [review]
LGTM, thanks everyone

**Pull request merged**
